### PR TITLE
add the ability to read from multiple Kafka topics

### PR DESCRIPTION
**Pull request by maha-hajja**
### Description

Changes the parameter "topic" from the source into a list of comma-separated topics to read from.

Fixes https://github.com/ConduitIO/conduit/issues/1482

Related to https://github.com/ConduitIO/conduit/issues/1471

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-kafka/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ A source is getting associated with a consumer group ID the first time the `Read
...
 |----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|---------------------------|
 | `servers`            | Servers is a list of Kafka bootstrap servers, which will be used to discover all the servers in a cluster.                                                                                                   | true     |                           |
-| `topic`              | Topic is the Kafka topic from which records will be read.                                                                                                                                                    | true     |                           |
+| `topic`              | Topic is a comma separated list of Kafka topics from which records will be read, ex: "topic1,topic1".                                                                                                        | true     |                           |
```
Comment (raulb) [inline README.md:32]: We should include `topics` here and add a mention regarding deprecating the use of `topic`. This connector is currently on 0.7.2. Maybe we could say that will be deprecated by 1.0.

File: source/config.go [hunk 2]
```diff
@@ -35,6 +41,23 @@ type Config struct {
...
+		return fmt.Errorf("required parameter missing: %q", "topics")
+	}
+	if len(c.Topic) > 0 && len(c.Topics) > 0 {
+		return fmt.Errorf(`can't provide both "topic" and "topics" params, "topic" will be deprecated soon, please only provide the "topics" parameter instead`)
+	}
+	if len(c.Topic) > 0 && len(c.Topics) == 0 {
+		sdk.Logger(ctx).Warn().Msg(`"topic" parameter will be deprecated soon, please use "topics" instead.`)
+		// add the topic value to the topics slice.
+		c.Topics = make([]string, 1)
+		c.Topics[0] = c.Topic
```
Comment (raulb) [inline source/config.go:60]: could we maybe actually respect all topics that were provided via `topic`?

e.g.: 

```suggestion
		c.Topics = make([]string, len(c.Topic))
		c.Topics = c.Topic
```

**Review by raulb — COMMENTED** [review]
@maha-hajja thanks for making the changes to start accepting `topics`! Left some comments. Let me know what you think.

File: source/config.go (hunk 2 above)
Comment (lovromazgon) [reply source/config.go:60]: `c.Topic` is just a single `string`, what do you mean by respect "all" topics? 🤔

File: source/config.go [hunk 3]
```diff
@@ -35,6 +41,23 @@ type Config struct {
 }
 
 // Validate executes manual validations beyond what is defined in struct tags.
-func (c Config) Validate() error {
-	return c.Config.Validate()
+func (c *Config) Validate(ctx context.Context) error {
+	err := c.Config.Validate()
+	if err != nil {
+		return err
+	}
```
Comment (lovromazgon) [inline source/config.go:48]: We should combine the errors coming from `c.Config.Validate` with the topic validation (see how we do it in `common.Config.Validate`).

File: test/util.go [hunk 4]
```diff
@@ -188,25 +197,27 @@ func CreateTopic(t T, servers []string, topic string) {
 	t.Cleanup(cl.Close)
 
 	adminCl := kadm.NewClient(cl)
-	resp, err := adminCl.CreateTopic(
-		ctx, 1, 1, nil, topic)
+	resp, err := adminCl.CreateTopics(
+		ctx, 1, 1, nil, topics...)
 	var kafkaErr *kerr.Error
-	if errors.As(err, &kafkaErr) && kafkaErr.Code == kerr.TopicAlreadyExists.Code {
+	if errors.As(resp.Error(), &kafkaErr) && kafkaErr.Code == kerr.TopicAlreadyExists.Code {
```
Comment (lovromazgon) [inline test/util.go:203]: Good catch 👍

File: source/config.go (hunk 2 above)
Comment (raulb) [reply source/config.go:60]: @lovromazgon oh, thought this was coming from yesterday's changes in which we had it as a slice (see https://github.com/ConduitIO/conduit-connector-kafka/pull/131/commits/3f258cad16df581013a0ec539b0785989ef6b22b#diff-2f3155aea603bde2c43d7e87f9fd14baf61f282ae3df1873fd70dd6456abc472R27). Didn't check today again that `Topic` had came back to its original type. Apologies for the confusion! @maha-hajja

File: README.md (hunk 1 above)
Comment (maha-hajja) [reply README.md]: oh sorry forgot about the readme! updated

File: README.md [hunk 5]
```diff
@@ -29,7 +29,8 @@ A source is getting associated with a consumer group ID the first time the `Read
...
 | `servers`            | Servers is a list of Kafka bootstrap servers, which will be used to discover all the servers in a cluster.                                                                                                   | true     |                           |
-| `topic`              | Topic is the Kafka topic from which records will be read.                                                                                                                                                    | true     |                           |
+| `topics`             | Topics is a comma separated list of Kafka topics from which records will be read, ex: "topic1,topic1".                                                                                                       | true     |                           |
+| ~~`topic`~~          | **{{will be deprecated by `v1.0`}}** Topic is the Kafka topic to read from.                                                                                                                                  | false    |                           |
```
Comment (lovromazgon) [inline README.md:33]: I'd change the deprecation warning (the field is already deprecated), move it to the end and point the user towards the field they should be using.
```suggestion
| `topics`             | Topics is a comma separated list of Kafka topics from which records will be read, ex: "topic1,topic2".                                                                                                       | true     |                           |
| ~~`topic`~~          | Topic is the Kafka topic to read from. **Deprecated: use `topics` instead.**                                                                                                                                 | false    |                           |
```

File: source/config.go [hunk 6]
```diff
@@ -35,6 +42,24 @@ type Config struct {
...
+	// validate and set the topics.
+	if len(c.Topic) == 0 && len(c.Topics) == 0 {
+		multierr = append(multierr, fmt.Errorf("required parameter missing: %q", "topics"))
+	}
+	if len(c.Topic) > 0 && len(c.Topics) > 0 {
+		multierr = append(multierr, fmt.Errorf(`can't provide both "topic" and "topics" params, "topic" will be deprecated soon, please only provide the "topics" parameter instead`))
+	}
+	if len(c.Topic) > 0 && len(c.Topics) == 0 {
+		sdk.Logger(ctx).Warn().Msg(`"topic" parameter will be deprecated soon, please use "topics" instead.`)
```
Comment (lovromazgon) [inline source/config.go:59]: On the same note, the field is already deprecated.
```suggestion
		sdk.Logger(ctx).Warn().Msg(`"topic" parameter is deprecated and will be removed, please use "topics" instead.`)
```

**Review by lovromazgon — APPROVED** [review]
Just some wording nitpicks, looks good otherwise 👍

File: source/config.go (hunk 6 above)
Comment (lovromazgon) [reply source/config.go:59]: Oh, and same in the error above in line 56.

**Pull request merged**